### PR TITLE
docs: document patches in envoy proxy

### PR DESCRIPTION
**Pull request by yaelSchechter**
**What type of PR is this?**
Document usage of patches (`StratigicMerge`) in EnvoyProxy.

### CODE REVIEW

File: site/content/en/latest/user/operations/customize-envoyproxy.md [hunk 1]
```diff
@@ -370,3 +370,43 @@ EOF
 [Gateway API documentation]: https://gateway-api.sigs.k8s.io/
 [EnvoyProxy]: ../../../api/extension_types#envoyproxy
 [egctl translate]: ../egctl/#validating-gateway-api-configuration
+
+## Customize EnvoyProxy with Patches
```
Comment (zirain) [inline site/content/en/latest/user/operations/customize-envoyproxy.md:374]: please put these before L370

Comment (yaelSchechter) [reply site/content/en/latest/user/operations/customize-envoyproxy.md:370]: thanks, changed

**Review by zirain — APPROVED** [review]
LGTM, thanks!

**Review by cnvergence — APPROVED** [review]

**Review by guydc — APPROVED** [review]

**Pull request merged**